### DISCUSSION TITLE
Is there a way to get wixobj file in wix4

### DISCUSSION THREAD

**Discussion by svg2003** [wixtoolset/issues#7807] [Questions]
In wix3, there was wixobj file, that can be used before to build the bundle.
We're using it to validate MSI, extract some information from there and prepare some files for the final bundle. 
But seems right now it's executed just as one action. Alternative to parse wxs myself (with all possible includes and libraries) very disappoints :(

**Comment by robmen** [comment]
Use the .wixpdb instead.

**Comment by svg2003** [comment]
It's generated after bundle is built, but we need this information before. For example, bundle must include SomethingBasedOnExistingMSIs.data file.

**Reply by robmen** [reply]
When MSIs build, they create .wixpdbs too.

**Comment by svg2003** [comment]
But how I can find which MSIs I need to use? I have only "wxs" from bundle project, and need somehow find all includes

**Reply by robmen** [reply]
Ahh, yeah. If you don't know the list of MSIs and want to solve this generically, you could parse the .wxs code or write a WixExtension to jump into the build pipeline. It's not a well-documented API, but neither were .wixobjs. Good luck.

**Comment by svg2003** [comment]
Is it feasible to introduce some option for wix.exe, to build only PDB without building final executable file? 
Or call some class/method from wix package to get the compiled wxs file?

**Reply by robmen** [reply]
> Is it feasible to introduce some option for wix.exe, to build only PDB without building final executable file?

Well, there are .wixipls. That *might* work for you. It's for pretty edge cases, but might work for yours.

> Or call some class/method from wix package to get the compiled wxs file?

That's what WixExtensions are.

**Comment by svg2003** [comment]
wixipls - is it some special output type? Thank you, I'll try to check what is it
And for WixExtensions - is the some sample project and/or documentation how to create them? Just in case - I need to be as less 'intrusive', as possible, i.e. it's very not desirable to add anything into wxs file, to call this extension. I can only use msbuild actions
